crates/ra_syntax/src/ast/builders.rs

```diff
@@ -3,41 +3,45 @@ use crate::ast::*;
 use crate::{AstNode, SmolStr, SyntaxKind, SyntaxTreeBuilder};
 use std::marker::PhantomData;
 
-pub trait AstNodeBuilder {
-    type Node: AstNode + Sized;
-
+pub trait AstMake {
```

Comment:
I've split  builder in two trait `AstMake` and `AstBuild`. This is made for implementing `token set` support. Token set can participate in ast generation but can't be converted to `AstNode` because it is not a node at all.